File: spring-web-java/src/main/java/spring/web/java/global/common/AESCryptoUtils.java
package spring.web.java.global.common;

import java.nio.charset.StandardCharsets;
import java.security.InvalidKeyException;
import java.security.NoSuchAlgorithmException;
import java.util.Base64;

import javax.crypto.BadPaddingException;
import javax.crypto.Cipher;
import javax.crypto.IllegalBlockSizeException;
import javax.crypto.NoSuchPaddingException;
import javax.crypto.SecretKey;
import javax.crypto.spec.SecretKeySpec;

public abstract class AESCryptoUtils {

	private final static String SECRET_KEY = "d9ANIqIyfTygI92m6jWFfAzUbEP73TNB";
	private final static String ALGORITHM = "AES";

	public static String encrypt(String plainText) {
		try {
			SecretKey secretKeySpec = getSecretKeySpec();
			Cipher cipher = getCipher(Cipher.ENCRYPT_MODE, secretKeySpec);
			byte[] encryptedBytes = cipher.doFinal(plainText.getBytes(StandardCharsets.UTF_8));
			return Base64.getEncoder().encodeToString(encryptedBytes);
		} catch (IllegalBlockSizeException | BadPaddingException e) {
			throw new RuntimeException(e);
		}
	}

	public static String decrypt(String encryptedText) {
		try {
			SecretKey secretKeySpec = getSecretKeySpec();
			Cipher cipher = getCipher(Cipher.DECRYPT_MODE, secretKeySpec);
			byte[] decryptedBytes = cipher.doFinal(Base64.getDecoder().decode(encryptedText));
			return new String(decryptedBytes, StandardCharsets.UTF_8);
		} catch (IllegalBlockSizeException | BadPaddingException e) {
			throw new RuntimeException(e);
		}
	}

	private static SecretKey getSecretKeySpec() {
		byte[] keyBytes = SECRET_KEY.getBytes(StandardCharsets.UTF_8);
		return new SecretKeySpec(keyBytes, ALGORITHM);
	}

	private static Cipher getCipher(int mode, SecretKey secretKeySpec) {
		Cipher cipher;
		try {
			cipher = Cipher.getInstance(ALGORITHM);
			cipher.init(mode, secretKeySpec);
		} catch (NoSuchAlgorithmException | NoSuchPaddingException | InvalidKeyException e) {
			throw new RuntimeException(e);
		}

		return cipher;
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/global/common/AESCryptoUtils.kt
package spring.web.kotlin.global.common

import java.nio.charset.StandardCharsets
import java.util.*
import javax.crypto.Cipher
import javax.crypto.spec.SecretKeySpec

class AESCryptoUtils(secretKey: String) {
    private val secretKeySpec: SecretKeySpec = SecretKeySpec(secretKey.toByteArray(StandardCharsets.UTF_8), "AES")

    fun encrypt(plainText: String): String = runCatching {
        val cipher = getCipher(Cipher.ENCRYPT_MODE, secretKeySpec)
        val encryptedBytes = cipher.doFinal(plainText.toByteArray())
        Base64.getEncoder().encodeToString(encryptedBytes)
    }.getOrElse { throw RuntimeException(it) }

    fun decrypt(encryptedText: String): String = runCatching {
        val cipher = getCipher(Cipher.DECRYPT_MODE, secretKeySpec)
        val decryptedBytes = cipher.doFinal(Base64.getDecoder().decode(encryptedText))
        String(decryptedBytes)
    }.getOrElse { throw RuntimeException(it) }

    private fun getCipher(mode: Int, secretKeySpec: SecretKeySpec) =
        Cipher.getInstance("AES").apply { init(mode, secretKeySpec) }
}
